apps/block_scout_web/lib/block_scout_web/generic_paging_options.ex

```diff
@@ -0,0 +1,78 @@
+defmodule BlockScoutWeb.GenericPagingOptions do
+  @moduledoc """
+  Defines generic paging options for paging by any key.
+  """
+
+  @type generic_paging_options :: {
+          order_field :: String.t() | nil,
+          order_dir :: String.t() | nil,
+          page_size :: non_neg_integer(),
+          page_number :: pos_integer()
+        }
+
+  def extract_paging_options_from_params(
+        params,
+        total_item_count,
+        allowed_order_fields,
+        default_order_dir,
+        default_page_size
+      ) do
+    page_size = extract_page_size(params, default_page_size)
+
+    %{
+      order_field: extract_order_field(params, allowed_order_fields),
+      order_dir: extract_order_dir(params, default_order_dir),
+      page_size: page_size,
+      page_number: extract_page_number(params, total_item_count, page_size)
+    }
+  end
+
+  defp extract_page_size(params, default_page_size) do
+    if Map.has_key?(params, "page_size") do
+      case Integer.parse(Map.get(params, "page_size", default_page_size)) do
+        {page_size, _} -> page_size
+        :error -> default_page_size
+      end
+    else
+      default_page_size
+    end
+  end
+
+  defp extract_page_number(params, total_item_count, page_size) do
+    if Map.has_key?(params, "page_number") do
```

Comment:
In general there's too much checking for the existence of a key before using it. In what situations are we trying to extract the page number when it does not exist? if we really need to we can handle this differently with a set of function clauses

``` elixir
defp extract_page_number(%{"page_number" => pg, total_item_count, page_size) do
 .....
end 
defp extract_page_number(_, _, _), do: 1

```